### PR TITLE
Fix a geonode provider layers fetching crasher

### PR DESCRIPTION
**Pull request by nirvn**
## Description

This PR fixes a crash in the geonode provider's layers fetching code. Long story short there is that everything was listening to the same signal, leading the individual layers parser request triggering an early end to the overall layers fetching process.

I've also fixed the URL string for individual layer fetching by appending a "/" at the end, to avoid needless redirect by the server (which comes at the cost of extra seconds).

### CODE REVIEW

**Comment by nirvn** [comment]
Unrelated hanna provider failure.

File: src/core/geocms/geonode/qgsgeonoderequest.cpp [hunk 1]
```diff
@@ -54,18 +54,22 @@ void QgsGeoNodeRequest::abort()
...
-    QList<QgsGeoNodeRequest::ServiceLayerDetail> layers;
-    if ( mError.isEmpty() )
+    if ( !mParsingLayers )
     {
-      layers = parseLayers( this->lastResponse() );
+      mParsingLayers = true;
+      QList<QgsGeoNodeRequest::ServiceLayerDetail> layers;
+      if ( mError.isEmpty() )
+      {
+        layers = parseLayers( this->lastResponse() );
```
Comment (elpaso) [inline src/core/geocms/geonode/qgsgeonoderequest.cpp:67]: is `this->` necessary?

Comment (nirvn) [reply src/core/geocms/geonode/qgsgeonoderequest.cpp:67]: @elpaso , most likely not, well spotted cleanup task :) I'll modify.

**Comment by nirvn** [comment]
@elpaso , done.

**Review by elpaso — APPROVED** [review]
LGTM

**Pull request merged**